proposals/stdlib/hex-format.md

```diff
@@ -0,0 +1,320 @@
+# HexFormat
+
+* **Type**: Standard Library API proposal
+* **Author**: Abduqodiri Qurbonzoda
+* **Status**: Implemented in Kotlin 1.9.0
+* **Prototype**: Implemented
+* **Target issue**: [KT-57762](https://youtrack.jetbrains.com/issue/KT-57762/)
+* **Discussion**: TBD
+
+## Summary
+
+Convenient API for formatting binary data into hexadecimal string form and parsing back.
+
+## Motivation
+
+Our research has shown that hexadecimal representation is more widely used than other numeric bases,
```

Comment:
Citation needed

Reply:
We don't have any public document with the results. But search in [grep.app](https://grep.app/search?q=toString%5C%28%28radix%20%3D%20%29%3F16%5C%29&regexp=true&filter[lang][0]=Kotlin) finds much more usages of `toString(16)` than any other base.

Reply:
Probably `.toString( /* 10 */ )` is more common, should we also consider introducing `DecFormat`?
I'm not against this proposal; it seems strange to tackle hexadecimal number formatting before decimal ones.

Reply:
The decimal representation of numeric values is indeed more common, and this proposal explicitly agrees with the proposition. However, this proposal is about introducing an API to facilitate formatting (and parsing) the hex representation of binary data. It provides use cases and proposes an API to make it easy to handle those use cases.
Of course, there are other potentially more impactful features. They deserve separate proposals.
`HexFormat` is not the most important or useful feature out there, but it addresses the pain points outlined in this document.

Reply:
Hi,
Discussion of the proposal has been moved here: https://github.com/Kotlin/KEEP/issues/362
Please don't hesitate to express your concerns there.